src/Command/Push/PushArtifactCommand.php

```diff
@@ -0,0 +1,284 @@
+    }
+
+    return 0;
+  }
+
+  /**
+   * Prepares a directory to build the artifact.
+   *
+   * @param \Closure $output_callback
+   * @param string $artifact_dir
+   * @param string $vcs_url
+   * @param string $vcs_path
+   *
+   * @throws \Acquia\Cli\Exception\AcquiaCliException
+   */
+  protected function prepareDir(Closure $output_callback, string $artifact_dir, string $vcs_url, string $vcs_path): void {
+    $fs = $this->localMachineHelper->getFilesystem();
+
+    $output_callback('out', "Removing $artifact_dir if it exists");
+    $fs->remove($artifact_dir);
+
+    $output_callback('out', "Initializing Git in $artifact_dir");
+    $process = $this->localMachineHelper->execute(['git', 'clone', '--depth', '1', '--branch', $vcs_path, $vcs_url, $artifact_dir], $output_callback, NULL, $this->output->isVerbose());
+    if (!$process->isSuccessful()) {
+      throw new AcquiaCliException('Failed to clone repository from the Cloud Platform: {message}', ['message' => $process->getErrorOutput()]);
+    }
+
+    $output_callback('out', 'Global .gitignore file is temporarily disabled during artifact builds.');
+    $this->localMachineHelper->execute(['git', 'config', '--local', 'core.excludesFile', 'false'], $output_callback, $artifact_dir, $this->output->isVerbose());
+    $this->localMachineHelper->execute(['git', 'config', '--local', 'core.fileMode', 'true'], $output_callback, $artifact_dir, $this->output->isVerbose());
+
+    // Vendor directories can be "corrupt" (i.e. missing scaffold files due to earlier sanitization) in ways that break composer install.
+    $output_callback('out', 'Removing vendor directories');
+    foreach ($this->vendorDirs($artifact_dir) as $vendor_directory) {
+      $fs->remove(Path::join($artifact_dir, $vendor_directory));
+    }
+  }
+
+  protected function build(Closure $output_callback, string $artifact_dir): void {
+    // @todo generate a deploy identifier
+    // @see https://git.drupalcode.org/project/drupal/-/blob/9.1.x/sites/default/default.settings.php#L295
+    $output_callback('out', "Mirroring source files from {$this->dir} to $artifact_dir");
+    $originFinder = $this->localMachineHelper->getFinder();
+    $originFinder->files()->in($this->dir)
+      // Include dot files like .htaccess.
+      ->ignoreDotFiles(FALSE)
+      // Ignore VCS ignored files (e.g. vendor) to speed up the mirror (Composer will restore them later).
+      ->ignoreVCSIgnored(TRUE);
+    $targetFinder = $this->localMachineHelper->getFinder();
+    $targetFinder->files()->in($artifact_dir)->ignoreDotFiles(FALSE);
+    $this->localMachineHelper->getFilesystem()->mirror($this->dir, $artifact_dir, $originFinder, ['override' => TRUE, 'delete' => TRUE], $targetFinder);
+
+    $output_callback('out', 'Installing Composer production dependencies');
+    $this->localMachineHelper->execute(['composer', 'install', '--no-dev', '--no-interaction', '--optimize-autoloader'], $output_callback, $artifact_dir, $this->output->isVerbose());
```

Comment:
This is another new system dependency

Reply:
Au contraire, we use it in `pull` and `new`. As with Git, I agree we'd ideally check for it, but we can't do that before fixing `commandExists()` (it's _super_ common for `composer` to be an alias, in fact Acquia recommends it as a workaround for Composer 1/2 issues)